### PR TITLE
Integrated fixed test.

### PR DESCRIPTION
**Pull request by dragomirtitian**

### CODE REVIEW

File: src/compiler/transformers/declarations/transform-project.ts [hunk 1]
```diff
@@ -111,31 +78,41 @@ export function emitDeclarationsForFile(sourceFile: SourceFile, options: Compile
...
     if (sourceMap) {
-        if (!writer.isAtStartOfLine()) writer.rawWrite(getNewLineCharacter(options));
+        if (!writer.isAtStartOfLine()) writer.writeLine();
         writer.writeComment(sourceMap.sourceMappingURL);
+        writer.writeLine();
     }
 
-    if (diagnostics.length > 0) {
-        throw new Error(`Cannot transform file '${sourceFile.fileName}' due to ${diagnostics.length} diagnostics`);
-    }
```
Comment (dragomirtitian) [inline src/compiler/transformers/declarations/transform-project.ts:130]: I removed the throw here. The compiler does not generally throw errors on invalid code.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/commonMissingSemicolons.d.ts.diff [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                                        ~~~~~~~~~~~~~
+ !!! error TS2705: An async function or method in ES5/ES3 requires the 'Promise' constructor.  Make sure you have a declaration for the 'Promise' constructor or include 'ES2015' in your '--lib' option.
+     asynd function myAsyncFunction2(): void {}
+@@ -227,8 +228,10 @@
+ !!! error TS2304: Cannot find name 'myFunction2'.
+                             ~
+ !!! error TS1005: ';' expected.
+     function function(): void { }
++            
++!!! error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/commonMissingSemicolons.d.ts.diff]: This is a weird error, I guess we can detect this case and not print errors, on it WDYT?

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/collisionCodeGenEnumWithEnumMemberConflict.d.ts.diff [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
++
++collisionCodeGenEnumWithEnumMemberConflict.ts(3,5): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
++
++
++==== collisionCodeGenEnumWithEnumMemberConflict.ts (1 errors) ====
++    enum Color {
++        Color,
++        Thing = Color
++        ~~~~~
++!!! error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/collisionCodeGenEnumWithEnumMemberConflict.d.ts.diff]: This looks wrong. Color should get assigned 0 by the `constEvaluate` and thus Thing should also be zero if I try to recall the logic.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/classMemberWithMissingIdentifier2.d.ts.diff [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+ 
+ 
+-==== classMemberWithMissingIdentifier2.ts (7 errors) ====
++==== classMemberWithMissingIdentifier2.ts (8 errors) ====
+     class C { 
+         public {[name:string]:VariableDeclaration};
+               
+ !!! error TS1146: Declaration expected.
++              
++!!! error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/classMemberWithMissingIdentifier2.d.ts.diff]: We seem to have a general problem on reporting around ill-formed syntax. What do you think is the right way to go?

File: src/compiler/transformers/declarations/emit-resolver.ts [hunk 5]
```diff
@@ -174,6 +174,47 @@ export function createEmitDeclarationResolver(file: SourceFile, host: IsolatedEm
...
+        if (isNumericLiteral(node)) {
+            return factory.createNumericLiteral(node.text, node.numericLiteralFlags);
+        }
+        if (isBigIntLiteral(node)) {
+            return factory.createBigIntLiteral(node.text);
+        }
+        if (isStringLiteralLike(node)) {
+            return factory.createStringLiteral(node.text);
+        }
+
```
Comment (h-joo) [inline src/compiler/transformers/declarations/emit-resolver.ts:187]: Nit: would it be better to make line breaks more consistent?

File: src/compiler/transformers/declarations/transform-project.ts [hunk 6]
```diff
@@ -15,82 +13,51 @@ import {
...
-    for (const file of files) {
-        transformer(file, host);
-    }
-}
-
-export function emitDeclarationsForFile(sourceFile: SourceFile, options: CompilerOptions) {
-    options.declaration = true;
-    options.declarationMap = true;
-    const emitHost = createEmitDeclarationHost(options, sys);
+export function transpileDeclaration(sourceFile: SourceFile, emitHost: IsolatedEmitHost) {
```
Comment (h-joo) [inline src/compiler/transformers/declarations/transform-project.ts:59]: Please mark this `@internal`. Every file reading in the `typescript.d.ts` in will fail due to `IsolatedEmitHost` being internal.

File: tests/baselines/reference/APILibCheck.errors.txt [hunk 7]
```diff
@@ -0,0 +1,34 @@
+typescript.d.ts(9832,112): error TS2304: Cannot find name 'IsolatedEmitHost'.
```
Comment (h-joo) [inline tests/baselines/reference/APILibCheck.errors.txt:1]: This should be gone after marking `transpileDeclaration` in transform-project.ts

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/FunctionPropertyAssignments3_es6.d.ts.diff [hunk 8]
```diff
@@ -0,0 +1,30 @@
+// [[Reason: undefined]] ////
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/FunctionPropertyAssignments3_es6.d.ts.diff]: Unnecessary compiler error  for Ill-form syntax.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/FunctionPropertyAssignments5_es6.d.ts.diff [hunk 9]
```diff
@@ -0,0 +1,30 @@
+// [[Reason: undefined]] ////
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/FunctionPropertyAssignments5_es6.d.ts.diff]: Error on ill-formed syntax.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/MemberFunctionDeclaration5_es6.d.ts.diff [hunk 10]
```diff
@@ -0,0 +1,32 @@
+// [[Reason: undefined]] ////
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/MemberFunctionDeclaration5_es6.d.ts.diff]: error on ill-formed syntax.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/MemberFunctionDeclaration6_es6.d.ts.diff [hunk 11]
```diff
@@ -0,0 +1,35 @@
+// [[Reason: undefined]] ////
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/MemberFunctionDeclaration6_es6.d.ts.diff]: error on ill-formed syntax.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/ambientEnumDeclaration1.d.ts.diff [hunk 12]
```diff
@@ -0,0 +1,40 @@
+// [[Reason: undefined]] ////
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/ambientEnumDeclaration1.d.ts.diff]: This shouldn't trigger errors, I should probably take a look.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/classMemberWithMissingIdentifier.d.ts.diff [hunk 13]
```diff
@@ -0,0 +1,36 @@
+// [[Reason: undefined]] ////
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/classMemberWithMissingIdentifier.d.ts.diff]: error on ill-formed syntax.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/constEnum1.d.ts.diff [hunk 14]
```diff
@@ -0,0 +1,51 @@
+// [[Reason: undefined]] ////
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/constEnum1.d.ts.diff]: Errors in Enum value evaluation

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/constEnumDeclarations.d.ts.diff [hunk 15]
```diff
@@ -0,0 +1,33 @@
+// [[Reason: undefined]] ////
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/constEnumDeclarations.d.ts.diff]: Enum: Should be possible.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/constEnumPropertyAccess1.d.ts.diff [hunk 16]
```diff
@@ -0,0 +1,37 @@
+// [[Reason: undefined]] ////
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/constEnumPropertyAccess1.d.ts.diff]: Enum issue, should be solvable.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/constEnumPropertyAccess2.d.ts.diff [hunk 17]
```diff
@@ -0,0 +1,40 @@
+// [[Reason: undefined]] ////
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/constEnumPropertyAccess2.d.ts.diff]: enum issue.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/constEnumPropertyAccess3.d.ts.diff [hunk 18]
```diff
@@ -0,0 +1,50 @@
+// [[Reason: undefined]] ////
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/constEnumPropertyAccess3.d.ts.diff]: Enum issue : `~` operator not supported?

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/constEnums.d.ts.diff [hunk 19]
```diff
@@ -0,0 +1,322 @@
+// [[Reason: undefined]] ////
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/constEnums.d.ts.diff]: enum issues.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/constantEnumAssert.d.ts.diff [hunk 20]
```diff
@@ -0,0 +1,38 @@
+// [[Reason: undefined]] ////
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/constantEnumAssert.d.ts.diff]: enum

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/constructorWithIncompleteTypeAnnotation.d.ts.diff [hunk 21]
```diff
@@ -0,0 +1,50 @@
+// [[Reason: undefined]] ////
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/constructorWithIncompleteTypeAnnotation.d.ts.diff]: triggering on ill-formed syntax.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/declarationEmitDefaultExportWithStaticAssignment.d.ts.diff [hunk 22]
```diff
@@ -0,0 +1,102 @@
...
++                            ~~~~~~~
++!!! error TS9009: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
++    Example.Foo = Foo
++    
++==== index2.ts (1 errors) ====
++    import {Foo} from './foo';
++    export {Foo};
++    export default function Example(): void {}
++                            ~~~~~~~
++!!! error TS9009: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/declarationEmitDefaultExportWithStaticAssignment.d.ts.diff:78]: I don't know if I want to argue about this now, but the location of the error seems misleading.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/declarationEmitDestructuringParameterProperties.d.ts.diff [hunk 23]
```diff
@@ -0,0 +1,105 @@
...
++declarationEmitDestructuringParameterProperties.ts(14,29): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
++declarationEmitDestructuringParameterProperties.ts(14,32): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
+ 
+ 
+-==== declarationEmitDestructuringParameterProperties.ts (3 errors) ====
++==== declarationEmitDestructuringParameterProperties.ts (12 errors) ====
+     class C1 {
+         constructor(public [x, y, z]: string[]) {
+                     ~~~~~~~~~~~~~~~~~~~~~~~~~~
+ !!! error TS1187: A parameter property may not be declared using a binding pattern.
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/declarationEmitDestructuringParameterProperties.d.ts.diff:68]: This shouldn't be an ID error as it has it's own error.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/declarationEmitExpandoPropertyPrivateName.d.ts.diff [hunk 24]
```diff
@@ -0,0 +1,38 @@
...
+ ==== a.ts (0 errors) ====
+     interface I {}
+     export function f(): I { return null as I; }
+-==== b.ts (1 errors) ====
++==== b.ts (2 errors) ====
+     import {f} from "./a";
+     
+     export function q(): void {}
++                    ~
++!!! error TS9009: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/declarationEmitExpandoPropertyPrivateName.d.ts.diff:33]: same - misleading error location.

File: tests/baselines/reference/isolated-declarations/original/tsc/constEnums.d.ts [hunk 25]
```diff
@@ -279,42 +279,24 @@ declare function foo2(e: I2.C.E): void;
...
-constEnums.ts(25,5): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
-constEnums.ts(26,5): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
-constEnums.ts(27,5): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
-constEnums.ts(28,5): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
-constEnums.ts(29,5): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
-constEnums.ts(30,5): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
 constEnums.ts(33,5): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
+constEnums.ts(33,10): error TS2474: const enum member initializers must be constant expressions.
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/original/tsc/constEnums.d.ts:286]: I guess this feature snuck in after merge?

File: tests/baselines/reference/isolated-declarations/original/diff/enumConstantMembers.d.ts.diff [hunk 26]
```diff
@@ -29,6 +29,66 @@
...
+     enum E1 {
+         a = 1,
+         b
+@@ -74,10 +80,16 @@
+         b = 2 / 0.0,
+         c = 1.0 / 0.0,
+         d = 0.0 / 0.0,
+         e = NaN,
++        ~
++!!! error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/original/diff/enumConstantMembers.d.ts.diff:62]: These errors are misleading, they have their own errors, I guess we shouldn't print errors in these cases.

File: tests/baselines/reference/isolated-declarations/original/diff/constEnums.d.ts.diff [hunk 27]
```diff
@@ -31,15 +16,34 @@
...
++==== constEnums.ts (13 errors) ====
+     const enum Enum1 {
+        A0 = 100,
+     }
+     
+@@ -179,8 +180,10 @@
+             export module C {
+                 export const enum E {
+                     V1 = 1,
+                     V2 = A.B.C.E.V1 | 100
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/original/diff/constEnums.d.ts.diff:43]: umm.... these shouldn't be common, but I guess we can support it? original code looks like : 
```
module A {
    export module B {
        export module C {
            export const enum E {
                V1 = 1,
                V2 = A.B.C.E.V1 | 100
            }
        }
    }
}
```

File: tests/baselines/reference/isolated-declarations/original/diff/constEnumErrors.d.ts.diff [hunk 28]
```diff
@@ -16,3 +16,35 @@
...
+                ~
+ !!! error TS2567: Enum declarations can only merge with namespace or other enum declarations.
+         A
+@@ -75,8 +76,10 @@
+     const enum E1 {
+         // illegal case
+         // forward reference to the element of the same enum
+         X = Y,
++        ~
++!!! error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/original/diff/constEnumErrors.d.ts.diff:46]: [enum] I feel that this shouldn't be an error. as it's a separate error.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/declarationInAmbientContext.d.ts.diff [hunk 29]
```diff
@@ -0,0 +1,37 @@
...
+ /// [Errors] ////
++
++declarationInAmbientContext.ts(1,14): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
++declarationInAmbientContext.ts(1,17): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
++declarationInAmbientContext.ts(2,14): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
++declarationInAmbientContext.ts(2,17): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
++
++
++==== declarationInAmbientContext.ts (4 errors) ====
++    declare var [a, b];  // Error, destructuring declaration not allowed in ambient context
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/declarationInAmbientContext.d.ts.diff]: These are their own errors, so we shouldn't print errors on it.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/enumConstantMemberWithString.d.ts.diff [hunk 30]
```diff
@@ -0,0 +1,84 @@
...
++enumConstantMemberWithString.ts(26,5): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
++enumConstantMemberWithString.ts(31,5): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
+ 
+ 
+-==== enumConstantMemberWithString.ts (2 errors) ====
++==== enumConstantMemberWithString.ts (11 errors) ====
+     enum T1 {
+         a = "1",
+         b = "1" + "2",
++        ~
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/enumConstantMemberWithString.d.ts.diff]: enum: string concatenation not working.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/underscoreEscapedNameInEnum.d.ts.diff [hunk 31]
```diff
@@ -0,0 +1,27 @@
...
++
+ /// [Errors] ////
++
++underscoreEscapedNameInEnum.ts(3,5): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
++
++
++==== underscoreEscapedNameInEnum.ts (1 errors) ====
++    enum E {
++        "__foo" = 1,
++        bar = E["__foo"] + 1
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/underscoreEscapedNameInEnum.d.ts.diff]: Should be supported if we want to be consistent with other enum expressions.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/staticsInAFunction.d.ts.diff [hunk 32]
```diff
@@ -0,0 +1,35 @@
...
+ staticsInAFunction.ts(2,11): error TS2582: Cannot find name 'test'. Do you need to install type definitions for a test runner? Try `npm i --save-dev @types/jest` or `npm i --save-dev @types/mocha`.
+ staticsInAFunction.ts(3,4): error TS1128: Declaration or statement expected.
+@@ -19,10 +20,12 @@
+ staticsInAFunction.ts(4,22): error TS2693: 'any' only refers to a type, but is being used as a value here.
+ staticsInAFunction.ts(4,26): error TS1005: ';' expected.
+ 
+ 
+-==== staticsInAFunction.ts (14 errors) ====
++==== staticsInAFunction.ts (15 errors) ====
+     function boo{
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/staticsInAFunction.d.ts.diff]: An error on its own - missing () for function declaration.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/expandoFunctionExpressionsWithDynamicNames.d.ts.diff [hunk 33]
```diff
@@ -0,0 +1,53 @@
...
++
++
++==== expandoFunctionExpressionsWithDynamicNames.ts (4 errors) ====
++    // https://github.com/microsoft/TypeScript/issues/54809
++    
++    const s = "X";
++    
++    export const expr = (): void => {}
++                 ~~~~
++!!! error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/expandoFunctionExpressionsWithDynamicNames.d.ts.diff:41]: This shouldn't have triggered as it does have type annotation on the RHS. 
1. empty parameter.
2. void as a return type.
Triggering TS9009 should be enough.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/jsxNamespaceImplicitImportJSXNamespaceFromConfigPickedOverGlobalOne(jsx=react-jsx).d.ts.diff [hunk 34]
```diff
@@ -0,0 +1,85 @@
...
++    
++      type IntrinsicElements = {
++        [K in keyof ReactJSXIntrinsicElements]: ReactJSXIntrinsicElements[K] & {
++          css?: string
++        }
++      }
++    }
++    
++==== /index.tsx (1 errors) ====
++    export const Comp = () => <div css="color: hotpink;"></div>;
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/jsxNamespaceImplicitImportJSXNamespaceFromConfigPickedOverGlobalOne(jsx=react-jsx).d.ts.diff]: I have less understanding in tsx -- I guess it's okay to report on these?

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/noImplicitAnyDestructuringVarDeclaration.d.ts.diff [hunk 35]
```diff
@@ -0,0 +1,99 @@
...
+ 
+ 
+-==== noImplicitAnyDestructuringVarDeclaration.ts (14 errors) ====
++==== noImplicitAnyDestructuringVarDeclaration.ts (21 errors) ====
+     var [a], {b}, c: any, d: any; // error
+         ~~~
+ !!! error TS1182: A destructuring declaration must have an initializer.
+          ~
+ !!! error TS7031: Binding element 'a' implicitly has an 'any' type.
++         ~
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/noImplicitAnyDestructuringVarDeclaration.d.ts.diff]: This is its own error - we should not report.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/mergedDeclarations2.d.ts.diff [hunk 36]
```diff
@@ -0,0 +1,38 @@
...
+ 
+-==== mergedDeclarations2.ts (1 errors) ====
++==== mergedDeclarations2.ts (2 errors) ====
+     enum Foo {
+         b
+     }
+     enum Foo {
+         a = b
++        ~
++!!! error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/mergedDeclarations2.d.ts.diff]: oh, we seems not to support merged enum value assignments?

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/negateOperatorInvalidOperations.d.ts.diff [hunk 37]
```diff
@@ -0,0 +1,44 @@
...
+ 
+ 
+-==== negateOperatorInvalidOperations.ts (11 errors) ====
++==== negateOperatorInvalidOperations.ts (12 errors) ====
+     // Unary operator -
+     
+     // operand before -
+     var NUMBER1 = var NUMBER: any-;  //expect error
++                 
++!!! error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
```
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/negateOperatorInvalidOperations.d.ts.diff]: invalid syntax.

**Review by h-joo — APPROVED** [review]

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/jsxNamespaceImplicitImportJSXNamespaceFromConfigPickedOverGlobalOne(jsx=react-jsx).d.ts.diff (hunk 34 above)
Comment (h-joo) [reply tests/baselines/reference/isolated-declarations/auto-fixed/diff/jsxNamespaceImplicitImportJSXNamespaceFromConfigPickedOverGlobalOne(jsx=react-jsx).d.ts.diff]: Please ignore - I just realized this is a return type inference which we do not support.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/mergedDeclarations2.d.ts.diff (hunk 36 above)
Comment (dragomirtitian) [reply tests/baselines/reference/isolated-declarations/auto-fixed/diff/mergedDeclarations2.d.ts.diff]: No we don't they can merge across files which is something we definitely don't want.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/negateOperatorInvalidOperations.d.ts.diff (hunk 37 above)
Comment (dragomirtitian) [reply tests/baselines/reference/isolated-declarations/auto-fixed/diff/negateOperatorInvalidOperations.d.ts.diff]: Yeah. This test is too invalid. I did not review all of them yet.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/noImplicitAnyDestructuringVarDeclaration.d.ts.diff (hunk 35 above)
Comment (dragomirtitian) [reply tests/baselines/reference/isolated-declarations/auto-fixed/diff/noImplicitAnyDestructuringVarDeclaration.d.ts.diff]: We don't really do a lot of correlation between errors.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/expandoFunctionExpressionsWithDynamicNames.d.ts.diff (hunk 33 above)
Comment (dragomirtitian) [reply tests/baselines/reference/isolated-declarations/auto-fixed/diff/expandoFunctionExpressionsWithDynamicNames.d.ts.diff:41]: Agreed. Made a node to fix in a follow on PER

File: src/compiler/transformers/declarations/transform-project.ts (hunk 6 above)
Comment (dragomirtitian) [reply src/compiler/transformers/declarations/transform-project.ts:59]: We should discuss this. Is this a fully internal API? I would rather mark `IsolatedEmitHost` as being public.

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/commonMissingSemicolons.d.ts.diff (hunk 2 above)
Comment (dragomirtitian) [reply tests/baselines/reference/isolated-declarations/auto-fixed/diff/commonMissingSemicolons.d.ts.diff]: I don't think there is a good way to do this unless we are explicitly redetecting the syntax errors which is not something I would actually want to do. I will make a note to ask MS as well

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/declarationEmitDefaultExportWithStaticAssignment.d.ts.diff (hunk 22 above)
Comment (dragomirtitian) [reply tests/baselines/reference/isolated-declarations/auto-fixed/diff/declarationEmitDefaultExportWithStaticAssignment.d.ts.diff:78]: Not on this PR, but we should look at improving this. Noted

File: tests/baselines/reference/isolated-declarations/original/tsc/constEnums.d.ts (hunk 25 above)
Comment (dragomirtitian) [reply tests/baselines/reference/isolated-declarations/original/tsc/constEnums.d.ts:286]: Yeah. I was working on this one.

File: tests/baselines/reference/isolated-declarations/original/diff/constEnums.d.ts.diff (hunk 27 above)
Comment (dragomirtitian) [reply tests/baselines/reference/isolated-declarations/original/diff/constEnums.d.ts.diff:43]: yes. we should porobably support it.

File: src/compiler/transformers/declarations/transform-project.ts (hunk 6 above)
Comment (h-joo) [reply src/compiler/transformers/declarations/transform-project.ts:59]: I guess that's fine for me.

**Pull request merged**